src/shared/sam/cli/samCliInvoker.ts

```diff
@@ -59,7 +56,7 @@ export class DefaultSamCliProcessInvoker implements SamCliProcessInvoker {
         const samCliLocation: string | undefined = this.context.cliConfig.getSamCliLocation()
         if (!samCliLocation) {
             const err = new Error('SAM CLI location not configured')
-            this.context.logger.error(err)
+            getLogger().error(err)
```

Comment:
All of the code removed in this PR were there just for this line.

Reply:
Looks like the old code was IoC style (or planned to be)?

Reply:
Mostly, however some of the things passed in weren't given enough thoughtful consideration.